### PR TITLE
Command parser changes to work with arguments that don't take values

### PR DESCRIPTION
**Pull request by liealex**

### CODE REVIEW

File: dependencies/adi/command_parser/command_parser.cpp [hunk 1]
```diff
@@ -28,6 +28,8 @@ void CommandParser::parseArguments(
     int argc, char *argv[],
     std::map<std::string, struct Argument> command_map) {
     int arg_number = 1;
+    
+    // Stores mandatory arguments and their locaiton
```
Comment (dNechita) [inline dependencies/adi/command_parser/command_parser.cpp:32]: typo (locaiton)

File: examples/data_collect/main.cpp [hunk 2]
```diff
@@ -83,18 +83,18 @@ static const char kUsagePublic[] =
...
+        {"-n", {"--n", false, "", "1", true}},
+        {"-m", {"--m", false, "", "0", true}},
+        {"-wt", {"--wt", false, "", "0", true}},
+        {"-ip", {"--ip", false, "", "", true}},
+        {"-fw", {"--fw", false, "", "", true}},
+        {"-fps", {"--fps", false, "", "", true}},
+        {"-ccb", {"--ccb", false, "", "", true}},
+        {"-s", {"--split", false, "", "", false}},
+        {"-config", {"-CONFIG", true, "last", "", true}}};
+
```
Comment (dNechita) [inline examples/data_collect/main.cpp:97]: extra tab

File: dependencies/adi/command_parser/command_parser.cpp [hunk 3]
```diff
@@ -65,25 +70,51 @@ void CommandParser::parseArguments(
...
+                    }
+                }
+                if(!argument_found){
+                    m_command_vector.push_back({argv[i], ""});
+                    arg_number++;
+                }
+            } else {
+                if (!command_map[argv[i]].has_value) {
+                    m_command_vector.push_back({argv[i], "true"});
+                    arg_number++;
```
Comment (dNechita) [inline dependencies/adi/command_parser/command_parser.cpp:110]: on both branches you are incrementing arg_number so no reason to write it twice

File: dependencies/adi/command_parser/command_parser.cpp [hunk 4]
```diff
@@ -65,25 +70,51 @@ void CommandParser::parseArguments(
...
+        } else if (i != argc - 1 &&
+                   is_argument == 0) { // Solves -arg/--arg -arg value
+            int is_long_arg = std::string(argv[i]).find("--");
+            if (is_long_arg != -1) {
+                bool argument_found = false;
+                for (const auto& entry : command_map) {
+                    if (entry.second.long_option == argv[i]) {
+                        if (!entry.second.has_value) {
+                            m_command_vector.push_back({argv[i], "true"});
+                            arg_number++;
```
Comment (dNechita) [inline dependencies/adi/command_parser/command_parser.cpp:92]: arg_numer++ and argument_found=true happen on both branches of the if statement

File: dependencies/adi/command_parser/command_parser.cpp (hunk 1 above)
Comment (liealex) [reply dependencies/adi/command_parser/command_parser.cpp]: Solved

File: examples/data_collect/main.cpp (hunk 2 above)
Comment (liealex) [reply examples/data_collect/main.cpp]: Solved

File: dependencies/adi/command_parser/command_parser.cpp (hunk 3 above)
Comment (liealex) [reply dependencies/adi/command_parser/command_parser.cpp]: Solved

File: dependencies/adi/command_parser/command_parser.cpp (hunk 4 above)
Comment (liealex) [reply dependencies/adi/command_parser/command_parser.cpp]: Solved

**Review by dNechita — APPROVED** [review]

**Pull request merged**